gateway/api/views.py

```diff
@@ -429,6 +429,11 @@ def stop(self, request, pk=None):  # pylint: disable=invalid-name,unused-argumen
                             except RuntimeInvalidStateError:
                                 logger.warning("cancel failed")
 
+                            if jobinstance.session_id:
```

Comment:
Does it cancel the session for each job?

Reply:
yes, it is might be redundant, but it is ok. There might be a case when 2 or more sessions created per serverless job, so it is better to safely cancel them all.

Reply:
OK. Does it return an error not an exception?

Reply:
it does not return anything. Down inside runtime client it returns `requests.Response`, but it is not propagated up

Reply:
Does the cancel session cancel the jobs in the session?  If so, does it cause "cancel failed" for each job except one in the session?

Reply:
cancel session cancels all jobs in a session

Reply:
no, it does not cause cancel failed